File: leetcode/462.最少移动次数使数组元素相等-ii.go
package main

import (
	"fmt"
	"sort"
)

/*
 * @lc app=leetcode.cn id=462 lang=golang
 *
 * [462] 最少移动次数使数组元素相等 II
 */

// @lc code=start
func minMoves2(nums []int) int {
	sort.Ints(nums)
	target := nums[len(nums)/2]
	res := 0
	for _, num := range nums {
		res += Abs(target - num)
	}
	return res
}

func Abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

// @lc code=end
func main() {
	fmt.Println(minMoves2([]int{1, 2, 3}))
	fmt.Println(minMoves2([]int{1, 10, 2, 9}))
}

File: leetcode/462.最少移动次数使数组元素相等-ii.py
#
# @lc app=leetcode.cn id=462 lang=python3
#
# [462] 最少移动次数使数组元素相等 II
#
from typing import List


# @lc code=start
class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        nums.sort()
        target = nums[len(nums) // 2]
        return sum(
            abs(target - num)
            for num in nums
        )


# @lc code=end
print(Solution().minMoves2([1, 2, 3]))
print(Solution().minMoves2([1, 10, 2, 9]))
